runtime/reconcilers/model.go

```diff
@@ -1258,40 +1287,97 @@ func (r *ModelReconciler) executeSingle(ctx context.Context, executor *wrappedMo
 		if err != nil {
 			return nil, err
 		}
-		stageDuration = stageResult.ExecDuration
+		finalResult.ExecDuration += stageDuration
+
+		return finalResult, nil
+	})
+}
 
-		// We change the inputProps to be the result properties of the stage step
```

Comment:
E.g. this comment seems to me like it added some useful context to explain what's happening